### DISCUSSION TITLE
flask subdomain url_for  static not working

### DISCUSSION THREAD

**Discussion by waymen** [pallets/flask#4805] [Q&A] [answered]
my template
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>www.example.com:8000</title>
    <link rel="stylesheet" href="{{ url_for('www.static', filename='w.css') }}">
</head>
<body>
    <h1>www.example.com:8080</h1>
</body>
</html>
```
why? 
<img width="584" alt="image" src="https://user-images.githubusercontent.com/16431201/188406970-94534915-1fc4-4f17-b1ea-aabd0af18853.png">

- Python version:  3.10
- Flask version:  2.1

**Comment by waymen** [comment]
i hope  /static/w.css. not  http://example.com:8080

**Comment by ThiefMaster** [comment, marked as answer]
For problems with your own code, provide a minimal but complete snippet to reproduce. Just a template w/o surrounding code isn't complete...

**Reply by waymen** [reply]
## my Blueprint is www_bp (using subdomain)
```
from flask import Blueprint, render_template, views

www_bp = Blueprint('www', __name__, template_folder='templates', static_folder='static')


class WWWIndexView(views.View):
    def dispatch_request(self):
        return render_template('www.html')

www_bp.add_url_rule('/', endpoint='www', view_func=WWWIndexView.as_view('www'), subdomain='www')

## My template is  www.html
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>www.example.com:8000</title>
    <link rel="stylesheet" href="{{ url_for('www.static', filename='w.css') }}">
</head>
<body>
    <h1>www.example.com:8080</h1>
</body>
</html>
```
but when i acccess http://www.example.com:8000.   url_for  error: 404

<img width="884" alt="image" src="https://user-images.githubusercontent.com/16431201/188408910-dd245e78-8650-4b59-95c7-c0feb09bfdb0.png">
